src/api/spec/models/workflow_spec.rb

```diff
@@ -272,6 +312,26 @@
       login user
     end
 
+    context 'with matching labels filter' do
```

Comment:
Why test this again in this context?

Reply:
The first context is for `#call` which only tests that the `step#call` is triggered for valid filters with labels.
The second context is for `#steps` which again checks the `step#call` is triggered but also checks `WorkflowArtifactsPerStep` are created. Since `label` is only a filter option, that's why I am using `with matching labels filter` and `with unmatched labels filter` for both

Reply:
But why test `WorkflowArtifactsPerStep` twice? You don't have any special artifacts right?

Reply:
`WorkflowArtifactsPerStep` is only tested once.  First example is just checking that the `#call` is successfully called on subject.